src/components/artsy.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react"
+import * as sharify from "sharify"
+
+/**
+ * The currently signed-in user.
+ */
+export interface CurrentUser extends sharify.CurrentUser {}
+
+/**
+ * The Artsy specific props injected by the higher-order component produced by `ContextConsumer`.
+ *
+ * @see {@link ContextProvider}
+ * @see {@link ContextConsumer}
+ *
+ * @todo Once this PR for `rest` https://github.com/Microsoft/TypeScript/pull/13470 lands we’ll be able to not make this
+ *       optional and simply remove it from the props that a component wrapped with the `ContextConsumer` HOC accepts.
+ */
+export interface ContextProps {
+  artsy?: Props
```

Comment:
> any reason not to set currentUser directly on the context instead of wrapping it?

I was thinking that there would be more data made available this way in the future. In retrospect it might have been a premature decision 🤔 

What direction do you lean?